src/go/rdctl/pkg/shutdown/shutdown.go

```diff
@@ -17,34 +17,40 @@ limitations under the License.
 package shutdown
 
 import (
-	"errors"
 	"fmt"
 	"os"
 	"os/exec"
 	"regexp"
 	"runtime"
 	"time"
 
-	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/factoryreset"
 	"github.com/sirupsen/logrus"
 )
 
-func FinishShutdown(waitForShutdown bool) error {
+type ShutdownInfo struct {
+	waitForShutdown bool
+}
+
+func NewShutdownInfo(waitForShutdown bool) *ShutdownInfo {
```

Comment:
We're making a struct now so we can carry `waitForShutdown` around, right? It might be cleaner to convert `waitForAppToDieOrKillIt` to use a struct instead (since it doesn't get us anything as an API exposed to other packages).

Reply:
I moved to this because at one point I needed to make `waitForShutdown` available to one of the kill routines, and didn't want to keep pushing that value through function calls. But, sure, I can make it local to the package.